compiler/damlc/tests/daml-test-files/ActionFail.daml

```diff
@@ -1,6 +1,8 @@
 -- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
 -- All rights reserved.
 
+-- @INFO Use catOptionals
+
```

Comment:
What are we testing here? And why is this test called `ActionFail`?

Reply:
That's lost in the mists of time but if I had to guess, it's that pattern match failures work correctly in the presence of `RebindableSyntax` (https://github.com/ghc-proposals/ghc-proposals/pull/168).